### PR TITLE
Replace occurrences of Set with List where ordering matters

### PR DESCRIPTION
**Pull request by jguz-pubnub**

### CODE REVIEW

File: pubnub-chat-api/api/pubnub-chat-api.api [hunk 1]
```diff
@@ -402,17 +402,17 @@ public final class com/pubnub/chat/config/PushNotificationsConfig {
 }
 
 public final class com/pubnub/chat/membership/MembersResponse {
-	public fun <init> (Lcom/pubnub/api/models/consumer/objects/PNPage$PNNext;Lcom/pubnub/api/models/consumer/objects/PNPage$PNPrev;IILjava/util/Set;)V
-	public final fun getMembers ()Ljava/util/Set;
+	public fun <init> (Lcom/pubnub/api/models/consumer/objects/PNPage$PNNext;Lcom/pubnub/api/models/consumer/objects/PNPage$PNPrev;IILjava/util/List;)V
```
Comment (jguz-pubnub) [inline pubnub-chat-api/api/pubnub-chat-api.api:405]: I got an error while running `jvmApiCheck` and got the hint in logs to execute `:pubnub-chat-api:apiDump`.
I ran it and that's why this file has changed. Is it OK?

File: pubnub-chat-api/api/pubnub-chat-api.api
Comment (jguz-pubnub) [inline pubnub-chat-api/api/pubnub-chat-api.api:1]: I got an error while running `jvmApiCheck` and got the hint in logs to execute `:pubnub-chat-api:apiDump`.
I ran it and that's why this file has changed. Is it OK?

File: pubnub-chat-impl/src/commonMain/kotlin/com/pubnub/chat/internal/ChatImpl.kt [hunk 2]
```diff
@@ -879,26 +879,26 @@ class ChatImpl(
                                     )
                                 }
                             emitEventFutures.awaitAll()
-                        }.then { setMembershipsResponse: PNChannelMembershipArrayResult ->
+                        }.then { membershipResponse: PNChannelMembershipArrayResult ->
```
Comment (marcin-cebo) [inline pubnub-chat-impl/src/commonMain/kotlin/com/pubnub/chat/internal/ChatImpl.kt:882]: Why name changed from setMembershipsResponse to membershipResponse ?

**Review by wkal-pubnub — APPROVED** [review]
I think that failing test is flaky, I've had problems with it before

File: pubnub-chat-impl/src/commonMain/kotlin/com/pubnub/chat/internal/ChatImpl.kt (hunk 2 above)
Comment (jguz-pubnub) [reply pubnub-chat-impl/src/commonMain/kotlin/com/pubnub/chat/internal/ChatImpl.kt:882]: Indeed, that makes no sense. I'll restore it to the previous value

**Pull request merged**